shortlister/model.py

```diff
@@ -30,6 +30,9 @@ class Applicant:
     scores: Dict[Criterion, str]
     notes: str
 
+    def __hash__(self):
+        return hash(self.cv)
+
```

Comment:
In Python dataclasses you can set the behaviour of the attributes to avoid having to write your own functions. We would like a few things:

1. Sort applicants using name
2. A hash function bashed on CV
3. A nice printable applicant details which can be used by tournament.

We can add `order=True` and `frozen=True` to the dataclass options, and specify the behaviour of each field using the `field()` method, which is used like this:

```
name: str = field(compare=True, hash=True)
```

This says the name attribute should be used to compare Applicant objects, and in the hash function.

There is also `field(repr=False, compare=False)` for attributes we don't want to print or use in comparison.

If you add the appropriate `field()` options to each of the Applicant dataclass attributes, you don't have to write your own `__hash__` or comparison methods.

Reply:
In the case of applicants, maybe it's easier to define a function for it rather than tweaking the field for every attribute? 

Some fields like "compare" is set to True by default so I think I would need to set every other field to False if I only want to sort by applicant name.

I found defining a function to be quite simple and works quite well, but I could be unaware about its downsides:
```python
# Print only these attributes of Applicant
    def __repr__(self):
        return f"{self.name}\n{self.scores}\n{self.notes}"
```
Output:
```
1. Michael Davis
{PhD: 'Excellent', Research software: 'Satisfactory', Best practices: 'Unsatisfactory'}
2. Emma Jones
{PhD: 'Satisfactory', Best practices: 'Excellent', Research software: 'Excellent'}
```

Reply:
That makes sense. Format it as if it is a standard dataclass repr: `f"Applicant(name={self.name}, scores=(self.scores), notes={self.notes})", so it can be used for usual printing outside of the tournament too.